pkg/alertmanager/alertmanager.go

```diff
@@ -236,9 +244,47 @@ func (am *Alertmanager) ApplyConfig(userID string, conf *config.Config) error {
 	return nil
 }
 
+// Pause running jobs in the alertmanager that are able to be restarted and sets
+// to inactive
+func (am *Alertmanager) Pause() {
+	// Set to inactive
+	am.activeMtx.Lock()
+	am.active = false
+	am.activeMtx.Unlock()
+
+	// Ensure inhibitor is set before being called
+	if am.inhibitor != nil {
+		am.inhibitor.Stop()
+	}
+
+	// Ensure dispatcher is set before being called
+	if am.dispatcher != nil {
+		am.dispatcher.Stop()
+	}
```

Comment:
Where are they restarted when we "unpause"? If this is confirmed to be a bug, I would suggest to:
1. Add an `Unpause()` function moving all the "unpause" logic there
2. Enrich the tests to cover this issue

Reply:
My understanding is that it is the `ApplyConfig` method that will "unpause" alert manager. It will also start new inhibitor and dispatcher.

This is not really obvious from the code though. Some comment at least would be nice.

Reply:
You're right Peter. Not obvious tho.

Reply:
I will add a comment explaining that.